### ISSUE TITLE
Decaying learning rate and other schedules

### ISSUE THREAD

**Issue by emjotde** [marian-nmt/marian-train#56]
Pass "reporter" around to optimizers and models in order to be able to implement decaying learning rates and other decays, for instance used for guided alignment. 

OpenNMT allows to mutiply the learning rate with a decay-factor for each new epoch or whenever the cost on the validation set does not improve by at least a given factor. 

Maybe we should also rename "reporter" if we use it for things like that. 

Options to be added would be for example:

```
--learning-rate-decay 0.7 --start-decay-epoch 5 
```

What happens at each epoch after epoch `5` is:

```
lr = lr * 0.7
```

**Comment by emjotde** [comment]
Done and looks good.
